### ISSUE TITLE
Use Component API, not Element's

### ISSUE THREAD

**Issue by heruan** [vaadin/vaadin-confirm-dialog-flow#51]
The API should use `Component`, not `Element`, as in all the other Flow components. 

Not sure why in `ConfirmDialog` this is different, but I guess these methods should take a `Component`, not `Element`:

- `setConfirmButton`
- `setRejectButton`
- `setCancelButton`
- `setHeader`
- `setText`

I'd expect to have this signature:

```java
var button = new Button();
dialog.setConfirmButton(button);
```
not this one
```java
var button = new Button();
dialog.setConfirmButton(button.getElement());
```

**Comment by alvarezguille** [comment]
Thanks @heruan for noticing this, we'll first add overloads using Component, and then maybe deprecate the Element methods before removing anything

**Comment by alexberazouski** [comment]
Fixed in #53
